build/Dockerfile

```diff
@@ -12,57 +12,58 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM registry.access.redhat.com/ubi9/python-311 as wheel
 
-ARG WHEEL_VERSION=""
-USER root
-RUN --mount=type=cache,target=/root/.cache/pip \
-    python -m pip install --upgrade pip && \
-    python -m pip install build
-COPY tuning tuning
-COPY .git .git
-COPY pyproject.toml pyproject.toml
-# build wheel if wheel version is empty else download the wheel from PyPi
-RUN if [[ -z "${WHEEL_VERSION}" ]]; \
-    then python -m build --wheel --outdir /tmp; \
-    else pip download fms-hf-tuning==${WHEEL_VERSION} --dest /tmp --only-binary=:all: --no-deps; \
-    fi && \
-    ls /tmp/*.whl >/tmp/bdist_name
-
-
-FROM registry.access.redhat.com/ubi9/ubi AS release
-
-ARG CUDA_VERSION=11.8.0
+## Global Args #################################################################
+ARG BASE_UBI_IMAGE_TAG=latest
 ARG USER=tuning
 ARG USER_UID=1000
-ARG SET_NUM_PROCESSES_TO_NUM_GPUS=True
+ARG PYTHON_VERSION=3.11
+ARG WHEEL_VERSION=""
+
+## Base Layer ##################################################################
+FROM registry.access.redhat.com/ubi9/ubi:${BASE_UBI_IMAGE_TAG} as base
 
-USER root
+ARG PYTHON_VERSION
+ARG USER
+ARG USER_UID
 
 RUN dnf remove -y --disableplugin=subscription-manager \
         subscription-manager \
-        # we install newer version of requests via pip
-        python3.11-requests \
-    && dnf install -y make \
-        # to help with debugging
-        procps \
+    && dnf install -y python${PYTHON_VERSION} procps \
+    && ln -s /usr/bin/python${PYTHON_VERSION} /bin/python \
+    && python -m ensurepip --upgrade \
+    && dnf update -y \
```

Comment:
@jbusche can you verify the change still works as intended ?

Reply:
@tedhtchang, looks to me that the change was to install `procps`?  But I am now seeing in the https://quay.io/repository/jbusche/fms-hf-tuning?tab=tags that the quay security scans are green (passed)

I've done a quick test of the image with this on my small OpenStack 4.14 cluster:
https://github.com/foundation-model-stack/fms-hf-tuning/wiki/Installing-and-Testing-OpenShift-fms%E2%80%90hf%E2%80%90tuning-Stack#6-testing
and it looks good.

On Devstage, I also did a `single GPU twitter fine-tuning` https://github.com/foundation-model-stack/fms-hf-tuning/wiki/SFT%E2%80%90Trainer%E2%80%90Image-Test-process#single-gpu-twitter-fine-tuning
which looked good.

I also did a `single GPU llama fine-tuning` https://github.com/foundation-model-stack/fms-hf-tuning/wiki/SFT%E2%80%90Trainer%E2%80%90Image-Test-process#single-gpu-llama-fine-tuning
Which looked good.

Lastly I tried a `single GPU llama LoRA tuning` https://github.com/foundation-model-stack/fms-hf-tuning/wiki/SFT%E2%80%90Trainer%E2%80%90Image-Test-process#single-gpu-llama-lora-tuning
and that looked good.

I can't get a job with 2 gpu to get out of pending though, so that's all the testing I can currently try.

Reply:
Thanks @jbusche This is great. Could you approve

Reply:
Nice testing Jim! I think we will have to sort out the wiki though as we are trying to consolidate the JSON configs. So there should be a better place for those configs. I have some PRs in flight already across our repos and working to make the Lora vs FT vs PT configs more clear.